src/main/java/org/jboss/xavier/integrations/route/ToBeanRouter.java

```diff
@@ -36,6 +36,7 @@ public void process(Exchange exchange) throws Exception {
                         // store the reply from the bean on the OUT message
                         exchange.getOut().setHeaders(headers);
                         exchange.getOut().setBody(exchange.getIn().getBody());
+                        exchange.getOut().setAttachments(exchange.getIn().getAttachments());
```

Comment:
what is this for?

Reply:
@mrizzi A camel message is made of 3 important parts: **Header, Body, and Attachments**. The objective of `ToBeanRouter.java` is just to add an extra Header without changing the original message. In order to be consistent, we need to preserve HEADER, BODY and ATTACHMENT.

This blog might help to understand the structure of an apache camel message: https://fizalihsan.github.io/technology/camel.html

Reply:
thanks for the answer: my doubt was about understanding the need for us to copy from in to out in these routes.
Let's discuss this later since it's OT for this PR.